CMakeLists.txt

```
@@ -173,8 +173,20 @@ if (NOT CMAKE_CUDA_COMPILER_ID STREQUAL "NVIDIA")
       which requires setting Clang as C and CXX compilers and leaving nvcc as CUDA compiler.")
 endif()
 
-# ; creates a list here
-set (PYTHON_VERSIONS "3.6;3.7;3.8;3.9;3.10;3.11")
+cmake_dependent_option(PYTHON_VERSIONS "Build Python bindings" ON
+                       "BUILD_PYTHON" OFF)
+if (NOT BUILD_PYTHON AND PYTHON_VERSIONS)
+  message(WARNING "The BUILD_PYTHON option is set to off and the PYTHON_VERSIONS variable is
+          provided. The latter will be ignored.")
+elseif (BUILD_PYTHON AND "${PYTHON_VERSIONS}" STREQUAL "ON")
+  set(PYTHON_VERSIONS "3.7;3.8;3.9;3.10;3.11")
+elseif (NOT ${PYTHON_VERSIONS})
+  set(PYTHON_VERSIONS "")
+endif ()
+
+if (BUILD_PYTHON)
+  message(STATUS "Building DALI for Python versions: ${PYTHON_VERSIONS}")
+endif ()
```

Is there a reason why we would need to have `PYTHON_VERSIONS` as a dependent option on `BUILD_PYTHON` and then set the former to the list of versions? My point is, wouldn't this code be sufficient?
```suggestion
if (NOT BUILD_PYTHON AND PYTHON_VERSIONS)
  message(WARNING "The BUILD_PYTHON option is set to off and the PYTHON_VERSIONS variable is
          provided. The latter will be ignored.")
elseif (BUILD_PYTHON AND NOT PYTHON_VERSIONS)
  set(PYTHON_VERSIONS "3.7;3.8;3.9;3.10;3.11")
endif ()

if (BUILD_PYTHON)
  message(STATUS "Building DALI for Python versions: ${PYTHON_VERSIONS}")
endif ()
```

---

How about now?